capstone/src/main/java/com/google/sps/servlets/DeletePollOptionServlet.java

```diff
@@ -70,8 +76,9 @@ private long setMaxVotesAndId(PreparedQuery results) {
    */
   private void deleteEntity(long maxVotedId, PreparedQuery results, DatastoreService datastore) {
     for (Entity optionEntity : results.asIterable()) {
-      if (optionEntity.getKey().getId() == maxVotedId) {
-        datastore.delete(optionEntity.getKey());
+      Key optionEntityKey = optionEntity.getKey();
+      if (optionEntityKey.getId() == maxVotedId) {
+        datastore.delete(optionEntityKey);
         break;
       }
```

Comment:
I think you can avoid iterating through the results by creating a Key object with the desired kind and id.
`Key optionKey = KeyFactory.createKey('Option', maxVotedId)`
Then you'll have to put the `datastore.delete(optionKey)` in a try-catch block because it could throw EntityNotFoundException. But this will be O(1) instead of O(n)